icinga2/icinga.go

```diff
@@ -50,6 +53,23 @@ type WebClient struct {
 	TLSConfig         *tls.Config
 }
 
+func (s *WebClient) GetClientConfig() WebClient {
+	return WebClient{
+		napping:           s.napping,
+		URL:               s.URL,
+		Username:          s.Username,
+		Password:          s.Password,
+		Debug:             s.Debug,
+		DisableKeepAlives: s.DisableKeepAlives,
+		Zone:              s.Zone,
+		TLSConfig:         s.TLSConfig,
+	}
+}
+
+func (s *MockClient) GetClientConfig() WebClient {
+	return WebClient{}
+}
```

Comment:
This seems odd, the implementation for `WebClient` essentially creates a copy of the object it's called on, while the implementation for `MockClient` returns an empty WebClient. To do this properly we should extract the actual client config into its own type, and either embed that type in both `WebClient` and `MockClient`, or simply construct that type from the client on which `GetClientConfig()` is called on.

Just returning a different struct is probably easier, something like:

```suggestion
type ClientConfig struct {
	URL               string
	Username          string
	Password          string
	Debug             bool
	DisableKeepAlives bool
	Zone              string
	TLSConfig         *tls.Config
}

func (s *WebClient) GetClientConfig() ClientConfig {
	return ClientConfig{
		URL:               s.URL,
		Username:          s.Username,
		Password:          s.Password,
		Debug:             s.Debug,
		DisableKeepAlives: s.DisableKeepAlives,
		Zone:              s.Zone,
		TLSConfig:         s.TLSConfig,
	}
}

func (s *MockClient) GetClientConfig() ClientConfig {
	return ClientConfig{}
}
```